brave/brave-core
# Add first page load time metric

Resolves https://github.com/brave/brave-browser/issues/40590



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Test 1:

0. Start with fresh profile.
1. Visit an internal page such as `brave://settings`. Ensure metric does not exist.
2. Access a normal web page. Ensure metric exists and the value is correct.
3. Close browser, advance system time by 4 days, start browser. Access another web page. Ensure metric value does not change.
4. Close browser, advance system time by another 4 days, start browser. Access another web page. Ensure metric value does not change.
5. Repeat steps 0 - 2, and advance the system time by different intervals (i.e. 15 min, 45 min) and restart the browser after performing step 1 and before performing step 2. Ensure that the metric value is correct, according to the interval used.

Test 2:

0. Start again with a fresh profile.
1. Close the browser and advance the system time by 8 days.
2. Access a normal web page. Ensure the metric does not exist.
